### PR TITLE
FEATURE: Use `\hyphenations` case-insensitive (like `\patterns`)

### PR DESCRIPTION
**Pull request by alexander-nitsche**
Hi @vanderlee, 

this patch handles syllables defined under the `\hyphenations` command case-insensitively. This corresponds to the handling of syllables defined under `\patterns`.

Greetings
Alex

Fixes: https://github.com/vanderlee/phpSyllable/issues/70

### CODE REVIEW

**Review by vanderlee — APPROVED** [review]

**Comment by vanderlee** [comment]
Hi @alexander-nitsche.
Looks good.
This PR is on your own fork so cannot merge.
Once done I'd like to have a look at performance for this, but it should be solid as is.

File: src/Syllable.php [hunk 1]
```diff
@@ -754,13 +754,28 @@ private function parseWord($word)
             return [$word];
         }
 
+        $wordLowerCased = mb_strtolower($word);
+
         // Is it a pre-hyphenated word?
-        if (isset($this->hyphenation[$word])) {
-            return mb_split('-', $this->hyphenation[$word]);
+        if (isset($this->hyphenation[$wordLowerCased])) {
```
Comment (vanderlee) [inline src/Syllable.php:760]: Maybe refactor the splitting algorithm into a new method for readability?

File: src/Syllable.php [hunk 2]
```diff
@@ -754,13 +754,28 @@ private function parseWord($word)
...
+            $parts = [];
+            $part = '';
+            for ($i = 0, $j = 0; $i < mb_strlen($this->hyphenation[$wordLowerCased]); $i++) {
```
Comment (vanderlee) [inline src/Syllable.php:763]: Minor performance tweaks;
Return of mb_strlen(...) is constant during loop.
Prefer ++precrement over postcrement++ where possible.
To the best of my knowledge, PHP is still pretty bad at optimizing these kind of things.

File: src/Syllable.php [hunk 3]
```diff
@@ -754,13 +754,28 @@ private function parseWord($word)
...
+                if (mb_substr($this->hyphenation[$wordLowerCased], $i, 1) !== '-') {
+                    $part .= mb_substr($word, $j++, 1);
+                } else {
+                    $parts[] = $part;
+                    $part = '';
+                }
+            }
+            if (!empty($part)) {
```
Comment (vanderlee) [inline src/Syllable.php:771]: `if ($part !== '')`

File: src/Syllable.php (hunk 1 above)
Comment (alexander-nitsche) [reply src/Syllable.php:758]: Done.

File: src/Syllable.php (hunk 2 above)
Comment (alexander-nitsche) [reply src/Syllable.php]: > Return of mb_strlen(...) is constant during loop.

Done.

> Prefer ++precrement over postcrement++ where possible.

Is aligned to similar code places.

File: src/Syllable.php (hunk 3 above)
Comment (alexander-nitsche) [reply src/Syllable.php]: It orients itself to similar code places. See for example the end of parseWordByPatterns(). Would adjust it for all places or leave it as is.

**Comment by alexander-nitsche** [comment]
There are 3 PRs in the pipeline, the first being https://github.com/vanderlee/phpSyllable/pull/68. If it is approved and merged by you, I would push the other two PRs for you to approve and merge as well.

**Comment by vanderlee** [comment]
@alexander-nitsche are you planning a PR into the main repo for this soon?

**Comment by alexander-nitsche** [comment]
Rebased on the latest master.

**Comment by alexander-nitsche** [comment]
Closed in favor of https://github.com/vanderlee/phpSyllable/pull/73.

**Pull request closed without merging**